examples/reading_events.rs

```diff
@@ -0,0 +1,188 @@
+        .await?;
+    // endregion read-from-stream
+
+    // region iterate-stream
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion iterate-stream
+
+    Ok(())
+}
+
+pub async fn read_from_stream_position(client: &Client) -> Result<()> {
+    // region read-from-position
+    let options = ReadStreamOptions::default().origin(Origin::Point(10));
+    let result = client.read_stream("some-stream", &options, 20).await?;
+    // endregion read-from-position
+
+    // region iterate-stream
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion iterate-stream
+
+    Ok(())
+}
+
+pub async fn read_stream_overriding_user_credentials(client: &Client) -> Result<()> {
+    // region overriding-user-credentials
+    let options = ReadStreamOptions::default()
+        .origin(Origin::Start)
+        .authenticated(Credentials::new("admin", "changeit"));
+
+    let result = client.read_stream("some-stream", &options, All).await?;
+    // endregion overriding-user-credentials
+    Ok(())
+}
+
+pub async fn read_from_stream_position_check(client: &Client) -> Result<()> {
+    // region checking-for-stream-presence
+    let options = ReadStreamOptions::default().origin(Origin::Point(10));
+
+    let result = client.read_stream("some-stream", &options, All).await?;
+
+    match result {
+        ReadResult::Ok(mut events) => {
+            while let Some(event) = events.try_next().await? {
+                let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+                println!("Event> {:?}", test_event);
+            }
+        }
+
+        ReadResult::StreamNotFound(stream_name) => {
+            println!("Stream not found: {}", stream_name);
+
+            return Ok(());
+        }
+    }
+    // endregion checking-for-stream-presence
+    Ok(())
+}
+
+pub async fn read_stream_backwards(client: &Client) -> Result<()> {
+    // region reading-backwards
+    let options = ReadStreamOptions::default().origin(Origin::End);
+    let result = client.read_stream("some-stream", &options, All).await?;
+
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion reading-backwards
+
+    Ok(())
+}
+
+pub async fn read_from_all_stream(client: &Client) -> Result<()> {
+    // region read-from-all-stream
+    let mut events = client.read_all(&Default::default(), All).await?;
```

Comment:
Same comment as above (https://github.com/EventStore/EventStoreDB-Client-Rust/pull/49/files#r572118928) about adding the explicit position and direction.

Reply:
Done